### PR TITLE
Support enum

### PR DESCRIPTION
**Pull request by kgoguevgoget**
The following changes to the code support enums in the protofile definiton. 
Tests for GET and POST are also supported.

We make use of the `pb.type()` function in the `MapMessage` function. This lets us retrieve more information about the actual object we are looking at during the enumeration of messages. 
Once we find that we have an `enum` message type then instead of going into an infinte loop we process it correctly as it is a special case. 

The example and documentation have also been fixed to reflect the changes to the code base.

### CODE REVIEW

**Comment by kgoguevgoget** [comment]
This PR should fix issue #18

File: example/helloworld.proto [hunk 1]
```diff
@@ -8,10 +8,17 @@ service Greeter {
...
 
+enum Color {
+  RED = 0;
+  BLUE = 1;
+  GREEN = 2;
+}
+
 message HelloRequest {
-  string name = 1;
+  string displayname = 1;
```
Comment (ysugimoto) [inline example/helloworld.proto]: should be lower camel case as `displayName`

Comment (kgoguevgoget) [reply example/helloworld.proto]: fixed with latest commit.

**Comment by kgoguevgoget** [comment]
@ysugimoto I fixed the PR with your latest comments if all is good with you then please go ahead and merge this into master and build a release :) thanks!

**Comment by ysugimoto** [comment]
@kgoguevgoget `v1.2.4` has been released as the latest! thanks 😄

**Pull request merged**